youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,136 @@
+                                                                             'videoInfo', fatal=False)), video_id)
+
+        # extract basic properties of video
+        title = (self._html_search_regex(r'<title>([^<]+?)</title>', webpage, 'title', default='') or
+                 self._html_search_meta('twitter:title', webpage, 'title', default='')).replace(" - VidLii",
+                                                                                                "") or self._html_search_regex(
+            r'<h1>(.+?)</h1>', webpage, 'title', default=None)
+
+        description = strip_or_none(get_element_by_id('des_text', webpage))
+
+        uploader = self._html_search_regex(
+            r'<div[^>]+class="wt_person"[^>]*>(?:[^<]+)<a href="/user/[^>]*?>([^<]*?)<|<img src="[^>]+?class=('
+            r'?:"avt2\s*"|\'avt2\s*\')[^>]+?alt=(?:"([^"]+?)"|\'([^\']+?)\')>', webpage, 'uploader', default=None,
+            fatal=False)
+
+        video_url = videoInfo_dict.get("src")
+
+        # get additional properties
+        uploader_url = urljoin('https://www.vidlii.com/user/', uploader)
+
+        # returns date as YYYYMMDD
+        upload_date = unified_strdate(
+            self._html_search_meta('datePublished', webpage, 'upload_date', default=None,
+                                   fatal=False) or self._html_search_regex(r'<date>(['r'^<]*?)</date>', webpage,
+                                                                           'upload_date',
+                                                                           default=None, fatal=False))
+
+        categories = self._html_search_regex(
+            r'<div>Category:\s*<\/div>[\s\r]*<div>[\s\r]*<a href="\/videos\?c=[^>]*>([^<]*?)<\/a>', webpage, 'categories',
+            default=None, fatal=False)
+        tags = re.findall(r'<a href="/results\?q=[^>]*>\s*([^<]*)</a>', webpage) or None
+        duration = int_or_none(
+            self._html_search_meta('video:duration', webpage, 'duration', default=False, fatal=False) or
+            videoInfo_dict.get("dur"))
+
+        view_count_fb = re.findall(r'<strong>([^<]*?)</strong>', get_element_by_class("w_views", webpage))
+        view_count_fb = view_count_fb[0] if view_count_fb else None
+        view_count = int_or_none(self._html_search_regex(r'Views:[^<]*<strong>([^<]*?)<\/strong>', webpage, 'view_count',
+                                                         default=None, fatal=False)) or int_or_none(view_count_fb)
+
+        comment_count = int_or_none(
+            self._html_search_regex(r'Comments:[^<]*<strong>([^<]*?)<\/strong>|<span[^>]+id="cmt_num"[^>]*>(['
+                                    r'^<]+?)<\/span>', webpage, 'comment_count',
+                                    default=None, fatal=False))
+
+        average_rating = float_or_none(
+            self._html_search_regex(r'{[\s\r]*\$\("#rateYo"\).rateYo\({[^}]*rating:\s*([^,]*?),[^}.]*}',
```

Comment:
Should be more relaxed. Any minor change HTML layout of the first part will break it.

Reply:
FIXED.